src/esp/assets/ResourceManager.cpp

```diff
@@ -384,6 +422,12 @@ bool ResourceManager::loadStage(
   if (!isSeparateSemanticScene) {
     flags |= RenderAssetInstanceCreationInfo::Flag::IsSemantic;
   }
+  // if texture-based semantic mesh specify in creation
+  // TODO: Currently do not support single mesh texture-based semantic and
+  // render stages
+  // if (renderInfo.hasSemanticTextures) {
+  //   flags |= RenderAssetInstanceCreationInfo::Flag::IsTextureBasedSemantic;
+  // }
```

Comment:
Just so I know, what's blocking this? As far as my understanding of the code here goes, this should be possible...

Reply:
It is possible, but, as currently implemented in habitat, if the renderAsset is somehow configured to also have semantic textures, the current process would not load the textures as needed to render them, but would rather attempt to convert all loaded textures to integers for semantic rendering.  Since, for now, this won't happen (we don't have render assets that also have semantic textures) I put this here mainly as a note for the future, when I refactor the loading pipeline.